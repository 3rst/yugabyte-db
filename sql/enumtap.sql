\unset ECHO
\i test_setup.sql

SELECT plan(72);
--SELECT * FROM no_plan();

-- This will be rolled back. :-)
SET client_min_messages = warning;
CREATE TYPE public.bug_status AS ENUM ('new', 'open', 'closed');
RESET client_min_messages;

/****************************************************************************/
-- Make sure has_type for enums.
SELECT * FROM check_test(
    has_type( 'bug_status' ),
    true,
    'has_type(enum)',
    'Type bug_status should exist',
    ''
);

/****************************************************************************/
-- Test has_enum().
SELECT * FROM check_test(
    has_enum( 'bug_status' ),
    true,
    'has_enum(enum)',
    'Enum bug_status should exist',
    ''
);

SELECT * FROM check_test(
    has_enum( 'bug_status', 'mydesc' ),
    true,
    'has_enum(enum, desc)',
    'mydesc',
    ''
);
SELECT * FROM check_test(
    has_enum( 'public'::name, 'bug_status'::name ),
    true,
    'has_enum(scheam, enum)',
    'Enum public.bug_status should exist',
    ''
);
SELECT * FROM check_test(
    has_enum( 'public', 'bug_status', 'mydesc' ),
    true,
    'has_enum(schema, enum, desc)',
    'mydesc',
    ''
);

-- Try failures.
SELECT * FROM check_test(
    has_enum( '__foobarbaz__' ),
    false,
    'has_enum(enum)',
    'Enum __foobarbaz__ should exist',
    ''
);
SELECT * FROM check_test(
    has_enum( '__foobarbaz__', 'mydesc' ),
    false,
    'has_enum(enum, desc)',
    'mydesc',
    ''
);
SELECT * FROM check_test(
    has_enum( 'public'::name, '__foobarbaz__'::name ),
    false,
    'has_enum(scheam, enum)',
    'Enum public.__foobarbaz__ should exist',
    ''
);
SELECT * FROM check_test(
    has_enum( 'public', '__foobarbaz__', 'mydesc' ),
    false,
    'has_enum(schema, enum, desc)',
    'mydesc',
    ''
);

/****************************************************************************/
-- Test hasnt_enum().
SELECT * FROM check_test(
    hasnt_enum( '__foobarbaz__' ),
    true,
    'hasnt_enum(enum)',
    'Enum __foobarbaz__ should not exist',
    ''
);
SELECT * FROM check_test(
    hasnt_enum( '__foobarbaz__', 'mydesc' ),
    true,
    'hasnt_enum(enum, desc)',
    'mydesc',
    ''
);
SELECT * FROM check_test(
    hasnt_enum( 'public'::name, '__foobarbaz__'::name ),
    true,
    'hasnt_enum(scheam, enum)',
    'Enum public.__foobarbaz__ should not exist',
    ''
);
SELECT * FROM check_test(
    hasnt_enum( 'public', '__foobarbaz__', 'mydesc' ),
    true,
    'hasnt_enum(schema, enum, desc)',
    'mydesc',
    ''
);

-- Try failures.
SELECT * FROM check_test(
    hasnt_enum( 'bug_status' ),
    false,
    'hasnt_enum(enum)',
    'Enum bug_status should not exist',
    ''
);
SELECT * FROM check_test(
    hasnt_enum( 'bug_status', 'mydesc' ),
    false,
    'hasnt_enum(enum, desc)',
    'mydesc',
    ''
);
SELECT * FROM check_test(
    hasnt_enum( 'public'::name, 'bug_status'::name ),
    false,
    'hasnt_enum(scheam, enum)',
    'Enum public.bug_status should not exist',
    ''
);
SELECT * FROM check_test(
    hasnt_enum( 'public', 'bug_status', 'mydesc' ),
    false,
    'hasnt_enum(schema, enum, desc)',
    'mydesc',
    ''
);

/****************************************************************************/
-- Test enum_has_labels().
SELECT * FROM check_test(
    enum_has_labels( 'public', 'bug_status', ARRAY['new', 'open', 'closed'], 'mydesc' ),
    true,
    'enum_has_labels(schema, enum, labels, desc)',
    'mydesc',
    ''
);
SELECT * FROM check_test(
    enum_has_labels( 'public', 'bug_status', ARRAY['new', 'open', 'closed'] ),
    true,
    'enum_has_labels(schema, enum, labels)',
    'Enum public.bug_status should have labels (new, open, closed)',
    ''
);
SELECT * FROM check_test(
    enum_has_labels( 'bug_status', ARRAY['new', 'open', 'closed'], 'mydesc' ),
    true,
    'enum_has_labels(enum, labels, desc)',
    'mydesc',
    ''
);
SELECT * FROM check_test(
    enum_has_labels( 'bug_status', ARRAY['new', 'open', 'closed'] ),
    true,
    'enum_has_labels(enum, labels)',
    'Enum bug_status should have labels (new, open, closed)',
    ''
);

-- Try failures.
SELECT * FROM check_test(
    enum_has_labels( 'public', 'bug_status', ARRAY['new', 'closed', 'open'], 'mydesc' ),
    false,
    'enum_has_labels(schema, enum, labels, desc) fail',
    'mydesc',
    '        have: {new,open,closed}
        want: {new,closed,open}'
);
SELECT * FROM check_test(
    enum_has_labels( 'public', 'bug_status', ARRAY['new', 'open', 'Closed'], 'mydesc' ),
    false,
    'enum_has_labels(schema, enum, labels, desc) fail',
    'mydesc',
    '        have: {new,open,closed}
        want: {new,open,Closed}'
);
SELECT * FROM check_test(
    enum_has_labels( 'bug_status', ARRAY['new', 'closed', 'open'], 'mydesc' ),
    false,
    'enum_has_labels(enum, labels, desc) fail',
    'mydesc',
    '        have: {new,open,closed}
        want: {new,closed,open}'
);

/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;
