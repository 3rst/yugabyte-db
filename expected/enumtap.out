\unset ECHO
1..72
ok 1 - has_type(enum) should pass
ok 2 - has_type(enum) should have the proper description
ok 3 - has_type(enum) should have the proper diagnostics
ok 4 - has_enum(enum) should pass
ok 5 - has_enum(enum) should have the proper description
ok 6 - has_enum(enum) should have the proper diagnostics
ok 7 - has_enum(enum, desc) should pass
ok 8 - has_enum(enum, desc) should have the proper description
ok 9 - has_enum(enum, desc) should have the proper diagnostics
ok 10 - has_enum(scheam, enum) should pass
ok 11 - has_enum(scheam, enum) should have the proper description
ok 12 - has_enum(scheam, enum) should have the proper diagnostics
ok 13 - has_enum(schema, enum, desc) should pass
ok 14 - has_enum(schema, enum, desc) should have the proper description
ok 15 - has_enum(schema, enum, desc) should have the proper diagnostics
ok 16 - has_enum(enum) should fail
ok 17 - has_enum(enum) should have the proper description
ok 18 - has_enum(enum) should have the proper diagnostics
ok 19 - has_enum(enum, desc) should fail
ok 20 - has_enum(enum, desc) should have the proper description
ok 21 - has_enum(enum, desc) should have the proper diagnostics
ok 22 - has_enum(scheam, enum) should fail
ok 23 - has_enum(scheam, enum) should have the proper description
ok 24 - has_enum(scheam, enum) should have the proper diagnostics
ok 25 - has_enum(schema, enum, desc) should fail
ok 26 - has_enum(schema, enum, desc) should have the proper description
ok 27 - has_enum(schema, enum, desc) should have the proper diagnostics
ok 28 - hasnt_enum(enum) should pass
ok 29 - hasnt_enum(enum) should have the proper description
ok 30 - hasnt_enum(enum) should have the proper diagnostics
ok 31 - hasnt_enum(enum, desc) should pass
ok 32 - hasnt_enum(enum, desc) should have the proper description
ok 33 - hasnt_enum(enum, desc) should have the proper diagnostics
ok 34 - hasnt_enum(scheam, enum) should pass
ok 35 - hasnt_enum(scheam, enum) should have the proper description
ok 36 - hasnt_enum(scheam, enum) should have the proper diagnostics
ok 37 - hasnt_enum(schema, enum, desc) should pass
ok 38 - hasnt_enum(schema, enum, desc) should have the proper description
ok 39 - hasnt_enum(schema, enum, desc) should have the proper diagnostics
ok 40 - hasnt_enum(enum) should fail
ok 41 - hasnt_enum(enum) should have the proper description
ok 42 - hasnt_enum(enum) should have the proper diagnostics
ok 43 - hasnt_enum(enum, desc) should fail
ok 44 - hasnt_enum(enum, desc) should have the proper description
ok 45 - hasnt_enum(enum, desc) should have the proper diagnostics
ok 46 - hasnt_enum(scheam, enum) should fail
ok 47 - hasnt_enum(scheam, enum) should have the proper description
ok 48 - hasnt_enum(scheam, enum) should have the proper diagnostics
ok 49 - hasnt_enum(schema, enum, desc) should fail
ok 50 - hasnt_enum(schema, enum, desc) should have the proper description
ok 51 - hasnt_enum(schema, enum, desc) should have the proper diagnostics
ok 52 - enum_has_labels(schema, enum, labels, desc) should pass
ok 53 - enum_has_labels(schema, enum, labels, desc) should have the proper description
ok 54 - enum_has_labels(schema, enum, labels, desc) should have the proper diagnostics
ok 55 - enum_has_labels(schema, enum, labels) should pass
ok 56 - enum_has_labels(schema, enum, labels) should have the proper description
ok 57 - enum_has_labels(schema, enum, labels) should have the proper diagnostics
ok 58 - enum_has_labels(enum, labels, desc) should pass
ok 59 - enum_has_labels(enum, labels, desc) should have the proper description
ok 60 - enum_has_labels(enum, labels, desc) should have the proper diagnostics
ok 61 - enum_has_labels(enum, labels) should pass
ok 62 - enum_has_labels(enum, labels) should have the proper description
ok 63 - enum_has_labels(enum, labels) should have the proper diagnostics
ok 64 - enum_has_labels(schema, enum, labels, desc) fail should fail
ok 65 - enum_has_labels(schema, enum, labels, desc) fail should have the proper description
ok 66 - enum_has_labels(schema, enum, labels, desc) fail should have the proper diagnostics
ok 67 - enum_has_labels(schema, enum, labels, desc) fail should fail
ok 68 - enum_has_labels(schema, enum, labels, desc) fail should have the proper description
ok 69 - enum_has_labels(schema, enum, labels, desc) fail should have the proper diagnostics
ok 70 - enum_has_labels(enum, labels, desc) fail should fail
ok 71 - enum_has_labels(enum, labels, desc) fail should have the proper description
ok 72 - enum_has_labels(enum, labels, desc) fail should have the proper diagnostics
